### PR TITLE
[1400] Add guard

### PR DESCRIPTION
**Pull request by tvararu**
### Context

This allows us to automatically run rspec on files that have changed.

### Guidance to review

This runs pretty slowly, about 3 seconds for a model spec with 2 assertions. Suggestions welcome on how to improve this.

### CODE REVIEW

**Comment by tvararu** [comment]
[Add guard to manage-courses-backend](https://trello.com/c/6uslCKWD/1400-add-guard-to-manage-courses-backend)

**Comment by heidar** [comment]
my hero

**Comment by misaka** [comment]
Just wanted to add rubocop to it.

**Comment by misaka** [comment]
I haven't managed to get `pry` to run nicely in guard, unfortunately. Anyone have any ideas? I'm currently trying to compile Ruby with readline.

File: Guardfile [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+    # Rails files
+    rails = dsl.rails(view_extensions: %w(erb))
+    dsl.watch_spec_files_for(rails.app_files)
+    dsl.watch_spec_files_for(rails.views)
+
+    watch(rails.controllers) do |m|
+      [
+        rspec.spec.call("routing/#{m[1]}_routing"),
+        rspec.spec.call("controllers/#{m[1]}_controller"),
+        rspec.spec.call("acceptance/#{m[1]}")
```
Comment (heidar) [inline Guardfile]: We probably want to change this to run `spec/requests/` but we don't map these files clearly. We could change that so we can run the relevant request specs every time we modify controllers?

File: Guardfile [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+      ]
+    end
+
+    # Rails config changes
+    watch(rails.spec_helper)     { rspec.spec_dir }
+    watch(rails.routes)          { "#{rspec.spec_dir}/routing" }
+    watch(rails.app_controller)  { "#{rspec.spec_dir}/controllers" }
+
+    # Capybara features specs
```
Comment (heidar) [inline Guardfile]: I don't think we have any of these

File: Guardfile (hunk 1 above)
Comment (misaka) [reply Guardfile]: Noticed this too. I'm wondering if we need to sort this now though, or can we slowly improve this as we go?

Comment (misaka) [reply Guardfile]: Also, seeing as this is just for us devs, could alternatively just make the changes you feel need to be done to add to this PR. ;)

File: Guardfile (hunk 2 above)
Comment (tvararu) [reply Guardfile]: done

**Comment by tvararu** [comment]
Have made this significantly faster by using spring.

**Comment by misaka** [comment]
+1 for `--no-interactions`!

**Pull request merged**